packages/material-ui/src/Skeleton/Skeleton.js

```diff
@@ -82,65 +82,63 @@ const SkeltonRoot = experimentalStyled(
     alpha(theme.palette.text.primary, theme.palette.mode === 'light' ? 0.11 : 0.13)};
   height: 1.2em;
 
-  & .${skeltonClasses.text} {
-    margin-top: 0;
-    margin-bottom: 0;
-    height: auto;
-    transform-origin: 0 55%;
-    transform: scale(1, 0.60);
-    border-radius: ${({ theme }) => {
+  margin-top: ${({ styleProps }) => styleProps.variant === 'text' && 0};
```

Comment:
```suggestion
  margin-top: ${({ styleProps }) => styleProps.variant === 'text' ? 0 : undefined};
```

you need to use this format as false is not considered the same as undefined